### PR TITLE
feat: support multiple domain names for domain

### PR DESCRIPTION
**Pull request by danipran**
Support multiple domain names for a single domain. Allows redirect rules to apply to multiple domain names. Intended for configurations that might have a www variation of the same domain, e.g. `mysite.test` & `www.mysite.test`.

### CODE REVIEW

File: redirect/admin.py [hunk 1]
```diff
@@ -58,9 +58,24 @@ class RedirectRuleAdmin(admin.ModelAdmin):
...
 
+class DomainNameInline(admin.TabularInline):
+    model = DomainName
+    extra = 0
+    min_num = 2
+
+
+@admin.display(description="Domain names")
+def domain_names(self):
+    return ", ".join(self.names.values_list("name", flat=True))
```
Comment (voneiden) [inline redirect/admin.py:68]: Suggesting prefetching the names in DomainAdmin

```python
class DomainAdmin(admin.ModelAdmin):
    def get_queryset(self, request):
        return super().get_queryset(request).prefetch_related("names")
```

Might be prefetch doesn't work for values_list ( :shrug: ) or it requires some extra black magic. So may have to change this to use `.all()`:

```suggestion
    return ", ".join(domain_name.name for domain_name in self.names.all())
```

**Review by voneiden — APPROVED** [review]
:+1: left one optional note about prefetching DomainNames

**Comment by tuomas777** [comment]
Now that there are `Domain.name` and `DomainName.name` I think reading the code can easily get pretty confusing especially for non-regulars. What about renaming `Domain.name` -> `Domain.display_name` ? 🤔

File: redirect/admin.py (hunk 1 above)
Comment (danipran) [reply redirect/admin.py:68]: Seems to work with `values_list`.

**Pull request merged**